docs/llm-hosting/bench/spectest.py

```
@@ -0,0 +1,47 @@
+      f"TG={u['completion_tokens']/el:.2f} tok/s")
+
+# The throughput number is only meaningful if the model actually quoted back the
+# source. GEN=400 cannot cover all 13 handlers, so a correct run is normally a
+# clean prefix; only an actual mismatch invalidates the measurement.
+expect = "\n".join(CODE.split("\n\n")[:13]).strip()
+got = d["choices"][0]["text"].strip()
+if got and expect.startswith(got):
+    state = "OK" if got == expect else "TRUNCATED"
+else:
+    state = "DIVERGED"
+n = next((i for i, (a, b) in enumerate(zip(expect, got)) if a != b), len(got))
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)